trade-finance-manager-api/src/v1/controllers/auth/auth-service/index.js

```diff
@@ -0,0 +1,46 @@
+const authProvider = require('../auth-provider');
+const tfmUser = require('./get-or-create-tfm-user');
+const { issueJwtAndUpdateUser } = require('./issue-jwt-and-update-user');
+
+/**
+ * processSsoRedirect
+ * Process SSO redirect.
+ * 1) Get Entra user.
+ * 2) Get TFM user. If no user exists, create a new one.
+ * 3) Populate Entra and TFM user data.
+ * 4) Issue a JWT and update TFM user data.
+ * 7) Get redirect URL
+ * 8) Return user data, token and redirect URL.
+ * @param {object} pkceCodes PKCE Codes object
+ * @param {object} authCodeRequest Auth code request
+ * @param {string} code authZ code
+ * @param {string} state MSAL state, containing a base64 encoded JSON object of state passed to Microsoft prior to redirect
+ * @returns {Promise<object | {}>} TFM user, token and redirect URL or empty object
+ */
```

Comment:
we should commonise this state object to avoid repeating ourselves with this comment